webapp/app/model/eligibility_data.py

```diff
@@ -483,25 +439,28 @@ class ForeignCountrySuccessEligibility(RecursiveDataModel):
     """
     has_no_other_income: Optional[OtherIncomeEligibilityData]
     foreign_country_eligibility: str
-    elster_registration_method_eligibility: Optional[str]
-    elster_abrufcode_eligibility: Optional[str]
+    user_a_has_elster_account_eligibility: str
+    user_b_has_elster_account_eligibility: Optional[str]
+    
+    @validator('user_b_has_elster_account_eligibility', always=True)
+    def has_user_b_elster_account_eligibility(cls,v, values):
+        user_a = values.get('user_a_has_elster_account_eligibility')
+        user_b = v
+        
+        try:
+            declarations_must_be_set_no(user_a)
+        except:
+            if user_b:
+                return declarations_must_be_set_no(user_b)
+            raise
+
+        return v if v else "" 
```

Comment:
I think we can make this more understandable by some renamings, comments and by just using the value itself instead of the (otherwise recommended) helper method `declarations_must_be_set_no`. What do you think?

```suggestion
        user_a_has_elster_account = values.get('user_a_has_elster_account_eligibility')
        user_b_has_elster_account = v
        
        # One person case
        if not user_b_has_elster_account:
            declarations_must_be_set_no(user_a_has_elster_account)
        # Two person case
        else:
            if user_a_has_elster_account == 'yes' and user_b_has_elster_account == 'yes':
                raise InvalidEligiblityError

        return v
```

Reply:
I am not so sure if that would be correct. Basically, I think that we should keep declaration_set_no, if that is what we want to test to make it easier to use that specific functionality